### ISSUE TITLE
Duplicated label "@com_google_protobuf//:protobuf_python" in the deps of a py_proto_library

### ISSUE THREAD

**Issue by epertoso** [protocolbuffers/protobuf#9688]
**What version of protobuf and what language are you using?**
Version: v3.20.0
Language: Python

**What operating system (Linux, Windows, ...) and version?**
GNU/Linux (Debian Buster, dockerized)

**What runtime / compiler are you using (e.g., python version or gcc version)**
Python 3.10.3

**What did you do?**
1. Build the `test_py` target in the minimal example given below (`$ bazel build :test_py`)

**What did you expect to see**
A successful build.

**What did you see instead?**
```
ERROR: /home/epertoso/test/BUILD:3:17: Label '@com_google_protobuf//:protobuf_python' is duplicated in the 'deps' attribute of rule 'test_py'
ERROR: error loading package '': Package '' contains errors
INFO: Elapsed time: 0.038s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**
We depend on gRPC, which in turns depend on [protoc-gen-validate](https://github.com/envoyproxy/protoc-gen-validate). We saw the error after updating protobuf to v3.20, the failing target is [validate_py](https://github.com/envoyproxy/protoc-gen-validate/blob/dfcdc5ea103dda467963fb7079e4df28debcfd28/validate/BUILD#L28).

The error was introduced by this PR: https://github.com/protocolbuffers/protobuf/pull/9187, as the test https://github.com/protocolbuffers/protobuf/blob/bc799d78f81115940eec953e2937245c70e3e6e4/protobuf.bzl#L549 fails because `py_libs` and `deps` contain strings, and not `Label`s.

**Minimal example**

`WORKSPACE`
```
workspace(name = "test")

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "com_google_protobuf",
    sha256 = "b07772d38ab07e55eca4d50f4b53da2d998bb221575c60a4f81100242d4b4889",
    strip_prefix = "protobuf-3.20.0",
    urls = ["https://github.com/protocolbuffers/protobuf/archive/refs/tags/v3.20.0.tar.gz"],
)

load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")

protobuf_deps()
```

`BUILD`

```
load("@com_google_protobuf//:protobuf.bzl", "py_proto_library")

py_proto_library(
    name = "test_py",
    srcs = ["test.proto"],
    deps = ["@com_google_protobuf//:protobuf_python"],
)
```

`test.proto`
```
syntax = "proto2";
package test;

message Foo { optional int32 bar = 1; }
```

**Comment by Wyverald** [comment]
This is a very tricky problem and indeed one I did not foresee. We can't do 1 since the strings passed in are not "absolute" labels, and unfortunately `str(some_label_object)` isn't always "absolute" either (cue years of tech debt). We can't do 2 as you said, but we should do some variant of it: we'd need some way to convert strings to labels but using the current package's context, essentially restoring a deprecated API that was removed (`Label(..., relative_to_caller_repository=True)`).
